cmd/gateway-gcs.go

```diff
@@ -294,8 +282,48 @@ func (l *gcsGateway) ListBuckets() ([]BucketInfo, error) {
 	return b, nil
 }
 
-// DeleteBucket delete a bucket on GCS
+// DeleteBucket delete a bucket on GCS.
 func (l *gcsGateway) DeleteBucket(bucket string) error {
+	itObject := l.client.Bucket(bucket).Objects(l.ctx, &storage.Query{Delimiter: slashSeparator, Versions: false})
+	// We list the bucket and if we find any objects we return BucketNotEmpty error. If we
+	// find only "minio.sys.temp/" then we remove it before deleting the bucket.
+	gcsMinioPathFound := false
+	nonGCSMinioPathFound := false
+	for {
```

Comment:
Unable to imagine how functional test will solve without seeing it.

Overall, I understand GCS gives in one format and minio converts into S3 format.  Why can't we have functions which accept GCS format data types and retuning S3 data types and having them unit tested?

In the same meaning, `IsBucketEmpty()` would be a function which accepts required data and returns expected result.  Note that the function is not supposed to make GCS calls.